### ISSUE TITLE
docker has go in its closure, causing a huge closure size increase

### ISSUE THREAD

**Issue by edolstra** [NixOS/nixpkgs#41152]
## Issue description

In NixOS 17.09, Docker was about 450 MB:
```
$ nix path-info -S /nix/store/02v51hjpvq06r0vwdr71p6ffns4mhp0g-docker-17.09.1-ce 
/nix/store/02v51hjpvq06r0vwdr71p6ffns4mhp0g-docker-17.09.1-ce     450773104
```
But in 18.03, it's 1338 MB:
```
$ nix path-info -S /nix/store/ji0h0kgayxrdy7aj0bxxxi8b5sc8mp6x-docker-18.03.1
/nix/store/ji0h0kgayxrdy7aj0bxxxi8b5sc8mp6x-docker-18.03.1-ce    1338573048
```
This appears to be mostly because `go` is now in docker-containerd's closure:
```
$ nix why-depends ./result /nix/store/3c0cq0bl24jlr6h13wbaxg2v1bb0vzf8-bazaar-2.7.0
/nix/store/4d41b0ixw85di0cih2956jhdznx71558-nixos-system-hagbard-18.03.git.5f2da7f
╚═══activate: …rapped..WRAPPER_PATH=/nix/store/fsmsnn5i7zj7ffss1rvwkycay2lwvl7y-system-path/bin:/nix/store/fsms…
    => /nix/store/fsmsnn5i7zj7ffss1rvwkycay2lwvl7y-system-path
    ╚═══bin/docker -> /nix/store/ji0h0kgayxrdy7aj0bxxxi8b5sc8mp6x-docker-18.03.1-ce/bin/docker
        => /nix/store/ji0h0kgayxrdy7aj0bxxxi8b5sc8mp6x-docker-18.03.1-ce
        ╚═══libexec/docker/docker-containerd -> /nix/store/839jh51k7lsb27bw7x4ig9md8kpk65sj-docker-containerd/bin/containerd
            => /nix/store/839jh51k7lsb27bw7x4ig9md8kpk65sj-docker-containerd
            ╚═══bin/containerd: …cannot close stream 0/nix/store/8gyv6sms33szh2w7wlbyqly3fkki690j-go-1.9.5/share/goacBalancerWrap…
                => /nix/store/8gyv6sms33szh2w7wlbyqly3fkki690j-go-1.9.5
                ╚═══share/go/bin/go: …yp-mercurial-4.5/bin:/nix/store/3c0cq0bl24jlr6h13wbaxg2v1bb0vzf8-bazaar-2.7.0/bin'${PATH:+':'}$P…
                    => /nix/store/3c0cq0bl24jlr6h13wbaxg2v1bb0vzf8-bazaar-2.7.0
```

**Comment by edolstra** [comment]
The closure size graph should help identify the responsible commits: https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.docker.x86_64-linux#tabs-charts

**Comment by Anton-Latukha** [comment]
Good day.

Solved after #41435 merge.

**Comment by Anton-Latukha** [comment]
Yep. Go closure increase is due to 1a9316e47d378d3a56e0d21fecaff017bb50e3c0.
